src/Symfony/Component/Validator/Constraints/AbstractStringContains.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ */
+abstract class AbstractStringContains extends Constraint
+{
+    public const SUBSTRING = 'substring';
+    public const WORD = 'word';
+
+    /**
+     * @var string|array
+     */
+    public $expected;
```

Comment:
As this class is extended by `StringContains` and `StringNotContains` I don't think `expected` fit well for the `StringNotContains` constraint, it is a bit confusing. Maybe, a name like `text` would be better ?

Reply:
Agree. Another option is to add `abstract protected function getText()` and then expose in subclasses public properties with desired names.

Reply:
@vudaltsov that's almost exactly what I'm doing, I'll push this week some work on it. Spoiler: there are in my local change 8 validators (`StringContains`, `ContainsWord`, `StartsWith`, `EndsWith` and related `Not`s) that will use it.